src/app/model/Rock.ts

```diff
@@ -0,0 +1,109 @@
+import { Board } from './Board';
+import { PieceMovement } from './PieceMovement';
+import { Side, Square } from './Types';
+
+export class Rock implements PieceMovement {
+  id: string;
+  side: Side;
+
+  constructor(id: string, side: Side) {
+    this.id = id;
+    this.side = side;
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [];
+    movesToGo.concat(this._goUp(position, board));
+    movesToGo.concat(this._goDown(position, board));
+    movesToGo.concat(this._goLeft(position, board));
+    movesToGo.concat(this._goRight(position, board));
+
+    return movesToGo;
+  }
+
+  _goUp(position: Square, board: Board): Square[] {
```

Comment:
TypeScript nie jest już tak upośledzony jak JS i mamy modyfikatory dostepu takie jak private. Zastouj tam gdzie możesz private i też pamiętaj o readonly. Jak da się to zamiast let użyj const. 

Im mniej mutowania / zmian wartości tym mniej możliwości do popełnienie błędu :)

Reply:
no taaaak, zupełnie zapomniałem o modyfikatorach dostępu!